lib/bootstrap/environment.rb

```diff
@@ -64,11 +89,23 @@ def pattern_path(path)
 if $0 == __FILE__
   LogStash::Bundler.setup!({:without => [:build, :development]})
   require ARGV.shift
+
+  if settings = YAML.parse(IO.read("settings.yml"))
+    settings = settings.to_ruby
```

Comment:
YAML.parse returns an awkward `Psych::Nodes::Document` object that can be navigated to obtain the structure of the yaml file. `to_ruby` just converts the structure into a simpler ruby object. in this case, a hash of hashes.